portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,29 +29,52 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<String> greetings = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> greetings = new ArrayList<String>();
-    greetings.add("Hey");
-    greetings.add("Hola");
-    greetings.add("Hello");
-
     // Convert greetings into a json string
     String json = convertToJson(greetings);
 
-    // Send the JSON as the response
+    ArrayList<String> tasks = new ArrayList<>();
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      String commentEx = (String) entity.getProperty("message");
+      tasks.add(commentEx);
+    }
+
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(convertToJson(tasks));
   }
 
   private String convertToJson(ArrayList<String> greetings) {
-    String json = greetings.get(0);
-    json += "\n ";
-    json += greetings.get(1);
-    json += "\n ";
-    json += greetings.get(2);
-
+    String json = "";
+    for (int i = 0; i < greetings.size(); i++) {
+      json += "\n" + greetings.get(i);
+    }
     return json;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("user-ans");
+    greetings.add(text);
```

Comment:
Why are you storing the users post in both the instance variable named `greetings` as well as in the datastore.  Ideally we could just put it in the datastore.